server/services/BpmnToSsotParsing/BpmnToSsotParsing.js

```diff
@@ -15,117 +15,127 @@ const ssotMetadataObj = SsotBaseObjects.metadataObj;
-                const inputVar = {}
-                inputVar.name = key
-                inputVar.value = parameterObj[key]
-                inputVar.requireUserInput = true
-                parameterArray.push(inputVar)
-            })
-            instructionElement.rpaParameters = parameterArray
-        }
-
-        instructionElement.outputVariable = ""
-    })
-    return elementsArray
-}
+  bpmnActivities.forEach((activity) => {
+    const instructionElement = elementsArray.find(
+      (element) => element.id === activity._attributes.id
+    );
+    instructionElement.name = activity._attributes.name;
+    instructionElement.type = 'INSTRUCTION';
+
+    if (activity._attributes['arkRPA:application']) {
+      instructionElement.rpaApplication =
+        activity._attributes['arkRPA:application'];
+      instructionElement.rpaTask = activity._attributes['arkRPA:task'];
+
+      const parameterArray = [];
+      const parameterObj = JSON.parse(activity._attributes['arkRPA:inputVars']);
+      Object.keys(parameterObj).forEach((key) => {
+        const inputVar = {};
+        inputVar.name = key;
+        inputVar.value = parameterObj[key];
+        inputVar.requireUserInput = true;
+        parameterArray.push(inputVar);
+      });
+      instructionElement.rpaParameters = parameterArray;
+    }
+
+    instructionElement.outputVariable = '';
+  });
+  return elementsArray;
+};
 
 /**
  * @description Enriches elements in the elementsArray that should be of type marker
  * @returns {Array}  Array of elements for single source of truth
  */
 const enrichMarkerElements = (elementsArray) => {
-    const eventRegularExpression = new RegExp("^Event_.*$")
-    elementsArray.forEach(element => {
-        if (eventRegularExpression.test(element.id)) {
-            element.type = "MARKER"
-        }
-    })
-    return elementsArray
-}
+  const eventRegularExpression = new RegExp('^Event_.*$');
+  elementsArray.forEach((element) => {
+    if (eventRegularExpression.test(element.id)) {
+      element.type = 'MARKER';
+    }
+  });
+  return elementsArray;
+};
 
 /**
  * @description Parses an JSON created from the xml of the bpmn model to the single source of truth
  * @returns {string} JSON that has to be put in single source of truth file
  */
 const parseBpmnToSsot = (bpmnJson) => {
-
-    const ssot = {}
-    ssot.robotMetadata = ssotMetadataObj
-
-    const flows = bpmnJson['bpmn2:definitions']['bpmn2:process']['bpmn2:sequenceFlow'];
-    let elementsArray = findElements(flows)
-    const bpmnActivities = bpmnJson['bpmn2:definitions']['bpmn2:process']['bpmn2:task'];
-    elementsArray = enrichInstructionElements(elementsArray, bpmnActivities)
-    elementsArray = enrichMarkerElements(elementsArray)
-
-    ssot.elements = elementsArray
-    return JSON.stringify(ssot, null, 2);
+  const ssot = {};
+  ssot['_id'] = ssotMetadataObj['_id'];
+  ssot.starterId = ssotMetadataObj.starterId;
+  ssot.robotName = ssotMetadataObj.robotName;
```

Comment:
please insert these fileds already in the object constructor, like the following:
```javascript
const ssot = {
  _id = ssotMetadataObj['_id'];
  starterId = ssotMetadataObj.starterId;
  robotName = ssotMetadataObj.robotName;
}
```
Additionally, as we do not have a metadataObj anymore, these changes should also be represented by the code. Please move the hardcoded metadata values from `baseObjects` here and delete everything concerning the metadataObject

```javascript
const ssot = {
  _id: '6045eccfa9a07940e5763f0b',
  starterId: 'exampleID',
  robotName: 'exampleRobot',
}
```

Reply:
✅